3rdparty/find_dependencies.cmake

```diff
@@ -1115,115 +1151,117 @@ if(BUILD_GUI)
-        if(MSVC OR (CMAKE_C_COMPILER_ID MATCHES ".*Clang" AND
-            CMAKE_CXX_COMPILER_ID MATCHES ".*Clang"
-            AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 7))
-            set(FILAMENT_C_COMPILER "${CMAKE_C_COMPILER}")
-            set(FILAMENT_CXX_COMPILER "${CMAKE_CXX_COMPILER}")
-        else()
-            message(STATUS "Filament can only be built with Clang >= 7")
-            # First, check default version, because the user may have configured
-            # a particular version as default for a reason.
-            find_program(CLANG_DEFAULT_CC NAMES clang)
-            find_program(CLANG_DEFAULT_CXX NAMES clang++)
-            if(CLANG_DEFAULT_CC AND CLANG_DEFAULT_CXX)
-                execute_process(COMMAND ${CLANG_DEFAULT_CXX} --version OUTPUT_VARIABLE clang_version)
-                if(clang_version MATCHES "clang version ([0-9]+)")
-                    if (CMAKE_MATCH_1 GREATER_EQUAL 7)
-                        message(STATUS "Using ${CLANG_DEFAULT_CXX} to build Filament")
-                        set(FILAMENT_C_COMPILER "${CLANG_DEFAULT_CC}")
-                        set(FILAMENT_CXX_COMPILER "${CLANG_DEFAULT_CXX}")
+        set(FILAMENT_RUNTIME_VER "")
+        if(BUILD_FILAMENT_FROM_SOURCE)
+            message(STATUS "Building third-party library Filament from source")
+            if(MSVC OR (CMAKE_C_COMPILER_ID MATCHES ".*Clang" AND
+                CMAKE_CXX_COMPILER_ID MATCHES ".*Clang"
+                AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 7))
+                set(FILAMENT_C_COMPILER "${CMAKE_C_COMPILER}")
+                set(FILAMENT_CXX_COMPILER "${CMAKE_CXX_COMPILER}")
+            else()
+                message(STATUS "Filament can only be built with Clang >= 7")
+                # First, check default version, because the user may have configured
+                # a particular version as default for a reason.
+                find_program(CLANG_DEFAULT_CC NAMES clang)
+                find_program(CLANG_DEFAULT_CXX NAMES clang++)
+                if(CLANG_DEFAULT_CC AND CLANG_DEFAULT_CXX)
+                    execute_process(COMMAND ${CLANG_DEFAULT_CXX} --version OUTPUT_VARIABLE clang_version)
+                    if(clang_version MATCHES "clang version ([0-9]+)")
+                        if (CMAKE_MATCH_1 GREATER_EQUAL 7)
+                            message(STATUS "Using ${CLANG_DEFAULT_CXX} to build Filament")
+                            set(FILAMENT_C_COMPILER "${CLANG_DEFAULT_CC}")
+                            set(FILAMENT_CXX_COMPILER "${CLANG_DEFAULT_CXX}")
+                        endif()
+                    endif()
+                endif()
+                # If the default version is not sufficient, look for some specific versions
+                if(NOT FILAMENT_C_COMPILER OR NOT FILAMENT_CXX_COMPILER)
+                    find_program(CLANG_VERSIONED_CC NAMES clang-12 clang-11 clang-10 clang-9 clang-8 clang-7)
```

Comment:
clang-13, clang-14 are out and clang-15 will be out soon. We should add them here.

Reply:
Sounds good. We can test and add them in a different PR.